src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+public class ShooterSubsystem extends SubsystemBase {
+  /** Creates a new ShooterSubsystem. */
+
+  private static final int deviceID = 2;
+  private CANSparkMax m_motor;
+  private SparkMaxPIDController m_pidController;
+  private RelativeEncoder m_encoder;
+  public double kP, kI, kD, kIz, kFF, kMaxOutput, kMinOutput, maxRPM;
+
+  public ShooterSubsystem(){
+    // initialize motor
+    m_motor = new CANSparkMax(deviceID, MotorType.kBrushless);
+    m_motor.restoreFactoryDefaults();
+
+    m_pidController = m_motor.getPIDController();
+
+    // Encoder object created to display position values
+    m_encoder = m_motor.getEncoder();
+
+    // PID coefficients
+    kP = 0.0001; 
+    kI = 0.000001;
+    kD = 0; 
+    kIz = 0; 
+    kFF = 0.000015; 
+    kMaxOutput = .2; 
+    kMinOutput = 0;
+    maxRPM = 5700;
+
+    // set PID coefficients
+    m_pidController.setP(kP);
+    m_pidController.setI(kI);
+    m_pidController.setD(kD);
+    m_pidController.setIZone(kIz);
+    m_pidController.setFF(kFF);
+    m_pidController.setOutputRange(kMinOutput, kMaxOutput);
+
+    // display PID coefficients on SmartDashboard
+    SmartDashboard.putNumber("P Gain", kP);
+    SmartDashboard.putNumber("I Gain", kI);
+    SmartDashboard.putNumber("D Gain", kD);
+    SmartDashboard.putNumber("I Zone", kIz);
+    SmartDashboard.putNumber("Feed Forward", kFF);
+    SmartDashboard.putNumber("Max Output", kMaxOutput);
+    SmartDashboard.putNumber("Min Output", kMinOutput);
+  }
+
+  @Override
+  public void periodic(){
+    // This method will be called once per scheduler run
+    // read PID coefficients from SmartDashboard
+    double p = SmartDashboard.getNumber("P Gain", 0);
+    double i = SmartDashboard.getNumber("I Gain", 0);
+    double d = SmartDashboard.getNumber("D Gain", 0);
+    double iz = SmartDashboard.getNumber("I Zone", 0);
+    double ff = SmartDashboard.getNumber("Feed Forward", 0);
+    double max = SmartDashboard.getNumber("Max Output", 0);
+    double min = SmartDashboard.getNumber("Min Output", 0);
```

Comment:
Unlikely to happen, but what happens if SmartDashboard is disconnected or fails to retrieve the numbers?
Would your pidController set all the values to 0?

Reply:
These live SmartDashboard numbers are just there for testing so I can change the numbers without changing the code and redeploying so they probably won't be on the final robot code. Great problem scenario though I will keep that in mind when getting numbers from SmartDashboard especially with the vision subsystem since that gets all of its values from the network tables on SmartDashboard. Thanks for the feedback